src/main.rs

```diff
@@ -1,3 +1,176 @@
+use core::time::Duration;
+use serialport::{available_ports, SerialPort};
+use std::io::Read;
+
+static INSTRUCTIONS_LM: &str = "Linux Module (probably) faulty, return to UniElec";
+static INSTRUCTIONS_BUTTON: &str = "Check button";
+
+fn open_serial_port(path: &str) -> Result<Box<dyn SerialPort>, Box<dyn std::error::Error>> {
```

Comment:
concrete error type should work here
```suggestion
fn open_serial_port(path: &str) -> Result<Box<dyn SerialPort>, serialport::Error> {
```